libraries/libfc/src/io/json.cpp

```diff
@@ -596,7 +596,7 @@ namespace fc
          {
               int64_t i = v.as_int64();
               if( format == json::output_formatting::stringify_large_ints_and_doubles &&
-                  i > 0xffffffff )
+                  (i * ((i>0)-(i<0)) > 0xffffffff))
```

Comment:
I find this construct rather odd and I am a bit surprised that compilers would not give a warning about bool used in asthmatic statement.

Seems like: `(i > 0xffffffff) || (i < -0xffffffff)` would be clearer.

Reply:
how about `std::abs(i) > 0xffffffff`
probably `cmath` should be included

Reply:
Casting to `uint64_t` should also do the job, i.e. `static_cast<uint64_t>(i) > 0xffffffff`

Reply:
Thank you, all great suggestions! Changed.